pkg/types/kustomization.go

```diff
@@ -248,3 +251,10 @@ type GeneratorOptions struct {
 	// resource contents.
 	DisableNameSuffixHash bool `json:"disableNameSuffixHash,omitempty" yaml:"disableNameSuffixHash,omitempty"`
 }
+
+// KVSource represents a KV plugin backend.
+type KVSource struct {
+	Type string   `json:"type,omitempty" yaml:"type,omitempty"`
```

Comment:
s/Type/PluginType/

No need to save characters here, since this field will soon have a default value.
Also, don't want to collide with text searches for the ultra common word Type